src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -0,0 +1,1564 @@
+                    }
+                    else if (_getMethod.LocalDeclaredReadOnly && _setMethod.LocalDeclaredReadOnly)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_DuplicatePropertyReadOnlyMods, location, this);
+                    }
+                    else if (this.IsAbstract)
+                    {
+                        // Check abstract property accessors are not private.
+                        CheckAbstractPropertyAccessorNotPrivate(_getMethod, diagnostics);
+                        CheckAbstractPropertyAccessorNotPrivate(_setMethod, diagnostics);
+                    }
+                }
+                else
+                {
+                    if (!this.IsOverride)
+                    {
+                        var accessor = _getMethod ?? _setMethod;
+                        if ((object)accessor != null)
+                        {
+                            // Check accessibility is not set on the one accessor.
+                            if (accessor.LocalAccessibility != Accessibility.NotApplicable)
+                            {
+                                diagnostics.Add(ErrorCode.ERR_AccessModMissingAccessor, location, this);
+                            }
+
+                            // Check that 'readonly' is not set on the one accessor.
+                            if (accessor.LocalDeclaredReadOnly)
+                            {
+                                diagnostics.Add(ErrorCode.ERR_ReadOnlyModMissingAccessor, location, this);
+                            }
+                        }
+                    }
+                }
+            }
+
+            if ((object)explicitlyImplementedProperty != null)
+            {
+                CheckExplicitImplementationAccessor(this.GetMethod, explicitlyImplementedProperty.GetMethod, explicitlyImplementedProperty, diagnostics);
+                CheckExplicitImplementationAccessor(this.SetMethod, explicitlyImplementedProperty.SetMethod, explicitlyImplementedProperty, diagnostics);
+            }
+
+            _explicitInterfaceImplementations =
+                (object)explicitlyImplementedProperty == null ?
+                    ImmutableArray<PropertySymbol>.Empty :
+                    ImmutableArray.Create(explicitlyImplementedProperty);
+
+            // get-only auto property should not override settable properties
+            if ((_propertyFlags & Flags.IsAutoProperty) != 0)
+            {
+                if (_setMethod is null && !this.IsReadOnly)
+                {
+                    diagnostics.Add(ErrorCode.ERR_AutoPropertyMustOverrideSet, location, this);
+                }
+
+                CheckForFieldTargetedAttribute(diagnostics);
+            }
+
+            CheckForBlockAndExpressionBody(syntax, diagnostics);
+        }
+
+#nullable enable
+
+        protected abstract TypeSyntax GetTypeSyntax(SyntaxNode syntax);
```

Comment:
>GetTypeSyntax [](start = 38, length = 13)

I think we should eliminate an assumption that there is a type syntax. We can ask for type location for error reporting, but we shouldn't on presence of TypeSyntax that we can take a look at. Example is an EqualityContract property, there is no type syntax to point to.